dask/dataframe/io/parquet/arrow.py

```diff
@@ -0,0 +1,344 @@
+                row_groups = [
+                    piece.get_metadata().row_group(0) for piece in pieces
+                ]
+                if dataset.metadata:
+                    names = dataset.metadata.schema.names
+                else:
+                    piece = pieces[0]
+                    md = piece.get_metadata()
+                    names = md.schema.names
+
+            stats = []
+            for row_group in row_groups:
+                s = {"num-rows": row_group.num_rows, "columns": []}
+                for i, name in enumerate(names):
+                    column = row_group.column(i)
+                    d = {"name": name}
+                    if column.statistics:
+                        col_field = schema.field_by_name(name)
+                        cs_min = column.statistics.min
+                        cs_max = column.statistics.max
+                        d.update(
+                            {
+                                "min": cs_min,
+                                "max": cs_max,
+                                "null_count": column.statistics.null_count,
+                            }
+                        )
+                    s["columns"].append(d)
+                stats.append(s)
+
+        else:
+            stats = None
+        # TODO: these are both sets, maybe not safe to zip over them
+        if dataset.partitions:
+            for name, partition_set in zip(
+                dataset.partitions.partition_names, dataset.partitions.levels
+            ):
+                if name in meta.columns:
+                    meta[name] = pd.Categorical(
+                        categories=partition_set.keys, values=[]
+                    )
+
+        return (
+            index,
+            meta,
+            stats,
+            [
+                {
+                    "piece": piece,
+                    "kwargs": {
+                        "partitions": dataset.partitions,
+                        "categories": categories,
+                    },
+                }
+                for piece in pieces
+            ],
+        )
+
+    @staticmethod
+    def initialize_write(
+        df,
+        fs,
+        path,
+        append=False,
+        partition_on=None,
+        ignore_divisions=False,
+        division_info=None,
+        **kwargs
+    ):
+        dataset = fmd = None
+        i_offset = 0
+        if append and division_info is None:
+            ignore_divisions = True
+        fs.mkdirs(path)
+
+        if append:
+            try:
+                # Allow append if there is a dataset.metadata object available
+                # (or maybe if there is pandas metadata available)
+                dataset = pq.ParquetDataset(
+                    path, filesystem=get_pyarrow_filesystem(fs)
+                )
+                if not dataset.metadata:
+                    raise NotImplementedError(
+                        "_metadata file needed to `append` "
```

Comment:
Right - this is certainly doable.  The temporary error was just because the _metadata is used to check the row-group statistics for the `ignore_divisions=False` case